pallets/loans/src/lib.rs

```diff
@@ -781,13 +774,61 @@ pub mod pallet {
 
 			Ok(Some(T::WeightInfo::update_portfolio_valuation(count)).into())
 		}
+
+		/// Transfer debt from one loan to another loan,
+		/// repaying from the first loan and borrowing the same amount from the
+		/// second loan. `from_loan_id` is the loan used to repay.
+		/// `to_loan_id` is the loan used to borrow.
+		/// The repaid and borrow amount must match.
+		#[pallet::weight(10_000_000)] //TODO
+		#[pallet::call_index(11)]
+		pub fn transfer_debt(
+			origin: OriginFor<T>,
+			pool_id: T::PoolId,
+			from_loan_id: T::LoanId,
+			to_loan_id: T::LoanId,
+			repaid_amount: RepaidPricingAmount<T>,
+			borrow_amount: PricingAmount<T>,
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+
+			ensure!(
+				from_loan_id != to_loan_id,
+				Error::<T>::TransferDebtToSameLoan
+			);
+
+			let repaid_amount = Self::repay_action(&who, pool_id, from_loan_id, &repaid_amount)?.0;
+
+			match &borrow_amount {
+				PricingAmount::Internal(value) => {
+					ensure!(
+						*value == repaid_amount.repaid_amount()?.total()?,
+						Error::<T>::TransferDebtAmountMismatched
+					)
+				}
+				PricingAmount::External(_external) => {
+					// TODO (1): handle it once slippage is added.
+					// TODO (2): if quantity is measured as a rate,
+					// then instead of using slippage we can use quantity with
+					// decimals.
+				}
+			}
+
+			let _count = Self::borrow_action(&who, pool_id, to_loan_id, &borrow_amount)?;
+
+			Self::deposit_event(Event::<T>::TransferDebt {
+				pool_id,
+				from_loan_id,
+				to_loan_id,
+				amount: repaid_amount.repaid_amount()?.total()?,
```

Comment:
Supposing we fix the previous issue, then, no matter if we choose here `repaid_amount` or `borrow_amount`, the total amount will always be the same.